### ISSUE TITLE
Usage with Vuex

### ISSUE THREAD

**Issue by Ldoppea** [konvajs/vue-konva#2]
Hi,

First, sorry if this is the wrong place as it is not an issue. Is there a better way to ask for support?

I would like to use vue-konva in addition to Vuex, but i'm not sure how to handle drag events.

If I use your readme.md sample and add 'draggable:true' to the **configeCircle** object. Then I'll be able to drag and move the v-circle component. But **configCircle** object will not be updated with new 'x' and 'y' values. This seems logic to me as vuejs wants us to use one way bindings.

Now if I add a Vuex store to serve the **configCircle** object and I update this object on **dragmove** event with a custom action. Then the new **configCircle** values will be propagated to the v-circle object **while he is still being dragged**.

This seems to work correctly, but is this a good practice?
I'm not comfortable enough with vuejs nor konva so I'm not able to answer this question.

What are your thoughts about this?

**Comment by rafaesc** [comment]
Hi @Ldoppea,

yes, it's an interesting question, actually, a [CoreShape](http://rafaelescala.com/vue-konva-doc/#coreshapes) can't mutate itself, always you have to mutate the **config prop** through of [Vue reactivity in depth](https://vuejs.org/v2/guide/reactivity.html), like this
https://github.com/rafaesc/vue-konva/blob/master/examples/components/Stars.vue#L43-L46

Or, you can mutate the  **config prop** with [**to** method](https://konvajs.github.io/api/Konva.Circle.html#to__anchor) of Konva
like this:
https://github.com/rafaesc/vue-konva/blob/master/examples/components/Stars.vue#L56-L63

When you will mutate that config prop, the layout will regenerate, including the CoreShape. Because as you say, is to use one way bindings, and when you dragmove the shape, you use the core of Konva and it shouldn't mutate the config prop because you couldn't control at what time the state is mutating, and it would be difficult to maintain the mutations.

I hope it helps you.

Regards!

**Comment by rafaesc** [comment]
If you need more help you can ask here. Regards

**Comment by Ldoppea** [comment]
Hi,

Thx for your answer, I still have a lot to learn about component lifecyle mechanisms 👍

**Comment by Ldoppea** [comment]
Hi,

I continue this post as a feedback on vue-konva/vuex cohabitation.

I'm still working on my project using Vuex with vue-konva. I'm not sure yet if this was a good idea as some basic konva mechanisms are made difficult to implement. But here are my observations :
👍 It is very easy to split the code in modules
👍 The application appearance really depends on the application state, there is very few unexpected behavior
👍 It is easy to have multiple elements following each other like if they are grouped, even if they are in different layers. You just need to listen the same position object from the Vuex Store.
( ~ ) Due to previous point, if you want those elements to follow each other on a drag&move, then you must commit a lot of Vuex Store mutations (one each time dragmove() is called). 
( ~ ) Images cannot be stored in the Vuex Store. So if you wan't to have a 'cache' system, you should create your own singleton with reference to loaded images and add their uid to the Vuex Store.
( ~ ) If you split elements in different vue modules, your only way to search nodes with .find() is to use node.parent.parent... until you get the main Stage and then use .find() on it.
👎 It seems not possible to move konva nodes to another layer (ex: like in handleDragstart in the Stars example). If their configuration is updated by the Vuex Store so the element will be recreated in the original layer resulting to 2 elements displayed at the same time.

In summary, this allow to easily develop your app in gobal, but some tasks are made more difficult or impossible. Also I think yo loose the high performance aspect of Konva.

**Comment by rafaesc** [comment]
Added a [playground to isolates complex components](http://rafaelescala.com/vue-konva/#/)

Thanks @Ldoppea for the vuex example
http://rafaelescala.com/vue-konva/#/vuex-example

you can see the source here
https://github.com/rafaesc/vue-konva/tree/master/examples

Regards!

**Comment by Ldoppea** [comment]
Hi,

Great! Thanks.

I didn't expect my demo to be added in the project, there are still some "garbage" code like `console.log` and unused objects. I'll try to clean it a bit 😃 

But first, there is a weird behavior on http://rafaelescala.com/vue-konva/#/vuex-example
I don't know why but after a dragend the object is not updated (text is still red and shadow is more opaque). Seems like the `dragLayer` is not cleared. But store is correct and vue component tree is correct and logs are showing in correct order, so I don't understand why.
When trying in local : 
Building https://github.com/rafaesc/vue-konva/tree/master/examples has the bug
Building https://codesandbox.io/s/2wl4m9p3ky with vue-konva@1.0.5 works correctly

I tried to compare both codes they seems to be identical. So this is weird.

**Comment by maxwarch** [comment]
Hello,
I have a similar problem.
I make a component that include the v-stage and different v-layer + v-image. I'm using vue-router. I serialize the data with toJSON() and restore on navigate, no problem.

The problem is when i want to save dataURL : when i'm on a page containing the component and i navigate to the next page, also containing the component, i can't save dataURL :
`konva.js:1083 Konva warning: Unable to get data URL. Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.`
When i refresh this page, i can save data URL.

Can you see the problem ?
I try to redraw the stage, $forceUpdate Vue and destroy stage on the previous page but nothing work.
Thanks for your help.
